pallets/utility/src/benchmarking.rs

```diff
@@ -0,0 +1,178 @@
+    // `T::OffChainSignature` to limit it.
+    let raw_signature: [u8; 64] = bob.sign(&call.encode()).to_bytes();
+    let encoded = MultiSignature::from(Signature::from_raw(raw_signature)).encode();
+    // Native execution can generate a hard-coded signature using the following code:
+    // ```ignore
+    //  let hex_encoded = hex::encode(&encoded);
+    //  frame_support::debug::info!("Bob nonce:{} encoded:{:?}", nonce, &hex_encoded);
+    //  ```
+
+    let signature = T::OffChainSignature::decode(&mut &encoded[..])
+        .expect("OffChainSignature cannot be decoded from a MultiSignature");
+
+    RelayTxSetup {
+        caller: alice.account,
+        target: bob.account,
+        signature,
+        call,
+    }
+}
+
+/// Relay transaction setup for WASM code.
+/// Public keys of `alice` and `bob` and the signature of the relayed call are generated in the native part, so WASM does not need to use the crypto stuff.
+///
+#[cfg(not(feature = "std"))]
+fn relay_tx_setup<T: Trait>() -> RelayTxSetup<T> {
+    // Keys generated
+    let alice_pk = hex!("6a4f597d1a0004ee6fd08622baf93fc350c048aeb8a6bf253208b1a536539333");
+    let bob_pk = hex!("8a2f30f00294ca72f2e9572263c8cf96695a1d9fffff3f8b0d49171a917d9f31");
+
+    // Create account from generated keys.
+    let alice = User::<T>::new_from_public(alice_pk);
+    let bob = User::<T>::new_from_public(bob_pk);
+
+    // Create the relayed call and its signature.
+    let call = make_transfer_calls::<T>(1, alice.account.clone(), 1)
+        .pop()
+        .unwrap();
+    let nonce: AuthorizationNonce = Module::<T>::nonce(bob.account.clone());
+    let call = UniqueCall::new(nonce, call);
+
+    //
+    let data = hex!("01aa3fb75dddaa9d1c058097aa10814a46a411192124a1970e21fc9547a075045090a318672bd44baa8f2069c4484ffc1b0e133011a80a1aaf2a484970bcffd987");
+    let signature = T::OffChainSignature::decode(&mut &data[..])
+        .expect("OffChainSignature cannot be decoded from a MultiSignature");
+
+    RelayTxSetup {
+        caller: alice.account,
+        target: bob.account,
+        signature,
+        call,
+    }
+}
+
+benchmarks! {
+    _ {}
+
+    batch {
+        let c in 1..MAX_CALLS;
+
+        let u = User::<T>::new("ALICE", 1);
+        let calls = make_calls::<T>(c);
+
+    }: _(u.origin, calls)
+    verify {
+        // NB In this case we are using `frame_system::Call::<T>::remark` which makes *no DB
+        // operations*. This helps us to fetch the DB read/write ops only from `batch` instead of
+        // its batched calls.
+        // So there is no way to verify it.
+        // The following cases use `balances::transfer` to be able to verify their outputs.
+    }
+
+    batch_transfer {
+        let c in 1..MAX_CALLS;
+
+        let sender = User::<T>::new("SENDER", 1);
+        let receiver = User::<T>::new("RECEIVER", 1);
+
+        let transfer_calls = make_transfer_calls::<T>(c, receiver.account.clone(), 500);
+    }: batch(sender.origin, transfer_calls)
```

Comment:
for benchmarks, let's only use the no-op system remark calls. We will manually add the weight of the extrinsic being called to the total weight.

Reply:
Ok, It makes sense.
BTW I've also kept `fn*_transfer` in order to be able to use `verify` section (team agreement).
In the output file, I've removed those `*_transfer` functions and added an extra comment about that.